crates/pera-core/src/authority.rs

```diff
@@ -1522,6 +1533,28 @@ impl AuthorityState {
         Ok(())
     }
 
+    async fn handle_mpc_events(
+        &self,
+        inner_temporary_store: &InnerTemporaryStore,
+        effects: &TransactionEffects,
+        epoch_store: &Arc<AuthorityPerEpochStore>,
+    ) -> anyhow::Result<()> {
+        if !self.is_validator(epoch_store) {
+            return Ok(());
+        }
```

Comment:
It feels to me like this should've been captured upstream. Is there a reason for (only) putting it here?

Reply:
AFAIK each full node & validator should execute each of the certificates in the consensus output, but only validators should care about signature-MPC related stuff, like those event, so we put it here. This is where Omer put it as well. Where exactly would you want to put it?

Reply:
I supposed there would be a point upstream that distinguished between full-nodes and validators. I could not tell you where would be a better place to put it. Let's leave it here.